pkg/build/controller/factory/factory.go

```diff
@@ -318,6 +417,59 @@ func (lw *imageStreamLW) Watch(resourceVersion string) (watch.Interface, error)
 	return lw.client.ImageStreams(kapi.NamespaceAll).Watch(labels.Everything(), fields.Everything(), resourceVersion)
 }
 
+// buildPodDeleteLW is a ListWatcher implementation that watches for Pods(that are associated with a Build) being deleted
+type buildPodDeleteLW struct {
+	client ControllerClient
+	store  cache.Store
+}
+
+// List lists all Pods associated with a Build.
+func (lw *buildPodDeleteLW) List() (runtime.Object, error) {
+	glog.V(5).Info("Checking for deleted build pods")
+	buildList, err := lw.client.Client.Builds(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
+	if err != nil {
+		glog.V(4).Infof("Failed to find any builds due to error %v", err)
+		return nil, err
+	}
+	for _, build := range buildList.Items {
+		glog.V(5).Infof("Found build %s/%s", build.Namespace, build.Name)
+		if build.Status != buildapi.BuildStatusRunning && build.Status != buildapi.BuildStatusPending && build.Status != buildapi.BuildStatusNew {
+			glog.V(5).Infof("Ignoring build %s/%s because it is complete", build.Namespace, build.Name)
+			continue
+		}
+		pod, err := lw.client.KubeClient.Pods(build.Namespace).Get(buildutil.GetBuildPodName(&build))
+		if err != nil && !kerrors.IsNotFound(err) {
+			glog.V(4).Infof("Error getting pod for build %s/%s: %v", build.Namespace, build.Name, err)
+			return nil, err
+		}
+		if (err != nil && kerrors.IsNotFound(err)) || pod.Labels[buildapi.BuildLabel] != build.Name {
+			pod = nil
+		}
+		if pod == nil {
+			deletedPod := &kapi.Pod{
+				ObjectMeta: kapi.ObjectMeta{
+					Name:      buildutil.GetBuildPodName(&build),
+					Namespace: build.Namespace,
+				},
+			}
+			glog.V(4).Infof("No build pod found for build %s/%s, sending delete event for pod", build.Namespace, build.Name)
```

Comment:
but i should note that for a v(4) message, i'm not going to lose a lot of sleep over such things.